src/core/ContactsManager.cpp

```diff
@@ -186,3 +186,13 @@ void ContactsManager::onUnreadCountChanged()
     emit unreadCountChanged(user, model->unreadCount());
 }
```

Comment:
Why do you have a signal here, listened to in QML, that calls back to get the count here, then goes back to QML, to call into C++ in MainWindow just to use MacExtras, instead of adding something like this after the emit (or even connected C++ side if you want to be super safe):

```
    int re = 0;
    foreach (ContactUser *u, pContacts) {
        if (u->conversation())
            re += u->conversation()->unreadCount();
    }
#ifdef Q_OS_MAC
    QtMac::setBadgeLabelText(QString::number(re));
#endif